validphys2/src/validphys/scripts/get_hepdata.py

```diff
@@ -0,0 +1,187 @@
+            metadata_dic = yaml.safe_load(stream_metadata)
+        return metadata_dic["hepdata"]
+
+    def write_tables(self, file: bytes, table_id: Optional[int] = None) -> None:
+        """
+        Write a given table into a file and save it into the disk afterwards.
+
+        Parameters
+        ----------
+            file: bytes
+                HepData table in bytes
+            table_id: int
+                Denote the `table_id`-th table
+        """
+        self.folder.mkdir(exist_ok=True)
+        ins_id = self.url.split("/")[-1]
+        filename = f"HEPData-{ins_id}-v{self.version}-Table_{table_id}"
+
+        # Write the tables as YAML files in rawdata folder
+        with open(f"{self.folder}/{filename}.yaml", "wb") as table_yaml:
+            table_yaml.write(file)
+
+    def check_downloaded_tables(self, table_numbers: list, vcheck: bool=False) -> None:
+        """
+        Check if the number of downloaded tables correspond to the list of tables
+        in the metadata.yaml file. This function should also be extended to include
+        more checks (something much more relevant than this).
+        """
+        nb_yaml_files_rawdata = sum(1 for _ in self.folder.glob("**/*.yaml")) - 1
+
+        if (nb_yaml_files_rawdata != len(table_numbers)):
+            raise DownloadFail("Some of the tables were not downloaded properly.")
+
+        # Check if the version in the hep-metadata and the online one is the same
+        if vcheck and Path(f"{self.folder}/hep-metadata-v{self.version}.yaml").is_file():
+            with open(f"{self.folder}/hep-metadata-v{self.version}.yaml", "r") as file:
+                hep_metadata = yaml.safe_load(file)
+            if hep_metadata["version"] != self.webversion and not self.force_download:
+                raise VersionMismatch("The local version is different from HepData.")
+
+    def download_heptables(self, table_numbers: list) -> None:
+        """
+        Download all the HepData tables specified in the metadata.
+        """
+        # Make sure that the list does not contain duplicates
+        table_numbers = list(dict.fromkeys(table_numbers))
+
+        for table in self.tables_dict:
+            # Check the current HepData table ID
+            table_id = int(table["name"].split()[-1])
+            if table_id in table_numbers:
+                table_dict = requests.get(table["@id"]).json()
+                # Loop over the different table formats. The following surely
+                # could be done in a much more efficient way in order to get
+                # rid of the recursive loops.
+                for url_tab in table_dict["distribution"]:
+                    if "YAML" in url_tab["description"]:
+                        get_reads_yaml = requests.get(url_tab["contentUrl"])
+                        self.write_tables(get_reads_yaml.content, table_id)
+                log.info(f"Table {table_id} downloaded and stored properly.")
```

Comment:
I don't understand this: Surely the table is only written when self.write_tables is called?